src/openvic-simulation/misc/Modifier.cpp

```diff
@@ -291,23 +303,35 @@ bool ModifierManager::load_triggered_modifiers(ast::NodeCPtr root) {
 key_value_callback_t ModifierManager::_modifier_effect_callback(
 	ModifierValue& modifier, key_value_callback_t default_callback, ModifierEffectValidator auto effect_validator
 ) const {
-
-	return [this, &modifier, default_callback, effect_validator](std::string_view key, ast::NodeCPtr value) -> bool {
-		ModifierEffect const* effect = get_modifier_effect_by_identifier(key);
-		if (effect != nullptr) {
-			if (effect_validator(*effect)) {
-				if (!modifier.values.contains(effect)) {
-					return expect_fixed_point(assign_variable_callback(modifier.values[effect]))(value);
-				} else {
-					Logger::error("Duplicate modifier effect: ", key);
-					return false;
-				}
+	std::function<bool(ModifierEffect const*, ast::NodeCPtr)> add_modifier_cb = [this, &modifier, effect_validator](ModifierEffect const* effect, ast::NodeCPtr value) -> bool {
+		if (effect_validator(*effect)) {
+			if (!modifier.values.contains(effect)) {
+				return expect_fixed_point(assign_variable_callback(modifier.values[effect]))(value);
 			} else {
-				Logger::error("Failed to validate modifier effect: ", key);
+				Logger::error("Duplicate modifier effect: ", effect->get_identifier());
 				return false;
 			}
+		} else {
+			Logger::error("Failed to validate modifier effect: ", effect->get_identifier());
+			return false;
 		}
-		return default_callback(key, value);
+	};
+
+	return [this, &modifier, default_callback, effect_validator, add_modifier_cb](std::string_view key, ast::NodeCPtr value) -> bool {
+		ModifierEffect const* effect = get_modifier_effect_by_identifier(key);
+		if (effect != nullptr && value->cast_to<ast::IdentifierNode>() != nullptr) {
+			return add_modifier_cb(effect, value);
+		} else if (complex_modifiers.contains(key) && value->cast_to<ast::AbstractListNode>() != nullptr) {
+			return expect_dictionary([this, &key, &add_modifier_cb, &default_callback](std::string_view identifier, ast::NodeCPtr node) -> bool {
+				std::string flat_identifier = std::string(key);
+				flat_identifier += "_";
+				flat_identifier += identifier;
+				ModifierEffect const* effect = get_modifier_effect_by_identifier(flat_identifier);
+				if(effect != nullptr) {
+					return add_modifier_cb(effect, node);
+				} else return false;
```

Comment:
Could you add an error message here including the `flat_identifier` which couldn't be found?